client/src/components/admin/SubmissionsPage/SurveySubmissionsChart.tsx

```diff
@@ -0,0 +1,224 @@
+          id: selectedQuestion.id,
+          options: selectedQuestion.options.map((option) => {
+            return {
+              id: option.id,
+              text: option.text[surveyLanguage],
+              count: questionAnswers.reduce(
+                (count, qa: AnswerEntry & { type: 'checkbox' | 'radio' }) => {
+                  return qa.value === option.id ||
+                    (Array.isArray(qa.value) && qa.value.includes(option.id))
+                    ? count + 1
+                    : count;
+                },
+                0,
+              ),
+            };
+          }),
+        };
+        break;
+      case 'numeric':
+      case 'slider':
+        base = {
+          id: selectedQuestion.id,
+          options: buildNumericRange(
+            { min: selectedQuestion.minValue, max: selectedQuestion.maxValue },
+            questionAnswers,
+          ).map((bucket, bucketIndex, buckets) => {
+            return {
+              id: bucket,
+              text: `${bucket} ${
+                buckets[1] - buckets[0] === 1
+                  ? ''
+                  : '\u2013 ' + (bucket + (buckets[1] - buckets[0]))
+              }`,
+              count: questionAnswers.reduce(
+                (count, qa: AnswerEntry & { type: 'slider' | 'numeric' }) => {
+                  if (qa.value == null || qa.value < bucket) {
+                    return count;
+                  }
+                  // Last bucket must include the upper limit
+                  if (bucketIndex === buckets.length - 1) {
+                    return qa.value <= bucket + buckets[1] - buckets[0]
+                      ? count + 1
+                      : count;
+                  }
+                  return qa.value < bucket + buckets[1] - buckets[0]
+                    ? count + 1
+                    : count;
+                },
+                0,
+              ),
+            };
+          }),
+        };
+        break;
+      default:
+    }
+
+    const optionCount = base?.options.length;
+    setChartWidth(
+      optionCount <= 3
+        ? 220
+        : 220 + Math.min(760, Math.log2(optionCount - 2) * 180),
+    );
+    return base;
+  }, [selectedQuestion, surveyLanguage]);
+
+  return answerData ? (
+    <ResponsiveContainer
+      minWidth={250}
+      height={340}
+      style={{
+        position: 'absolute',
+        backgroundColor: '#ffffffdd',
+        borderBottomRightRadius: '7px',
+        maxWidth: `${chartWidth}px`,
+      }}
+    >
+      <BarChart
+        data={answerData.options}
+        margin={{
+          top: 25,
+          right: 30,
+          left: 20,
+          bottom: 60,
+        }}
+      >
+        <CartesianGrid strokeDasharray="3 3" />
+        <XAxis dataKey="text" tick={<CustomizedAxisTick />} interval={0} />
+        <YAxis />
+        <Tooltip />
+        <Bar dataKey="count" fill={theme.palette.primary.main} />
+      </BarChart>
+    </ResponsiveContainer>
+  ) : (
+    <></>
```

Comment:
[nit] I think you can also return `null` to denote empty contents, but I guess an empty fragment is pretty equivalent to this